libafl/src/stages/calibrate.rs

```diff
@@ -0,0 +1,211 @@
+use crate::{
+    bolts::current_time,
+    corpus::{Corpus, PowerScheduleTestData},
+    executors::{Executor, HasObservers},
+    fuzzer::Evaluator,
+    inputs::Input,
+    observers::{MapObserver, ObserversTuple},
+    stages::Stage,
+    state::{HasCorpus, HasMetadata},
+    Error,
+};
+use alloc::string::{String, ToString};
+use core::{marker::PhantomData, time::Duration};
+use num::Integer;
+use serde::{Deserialize, Serialize};
+
+/// The default mutational stage
+#[derive(Clone, Debug)]
+pub struct CalibrationStage<C, E, EM, I, O, OT, S, T, Z>
+where
+    T: Integer + Default + Copy + 'static + serde::Serialize + serde::de::DeserializeOwned,
+    C: Corpus<I>,
+    E: Executor<EM, I, S, Z> + HasObservers<I, OT, S>,
+    I: Input,
+    O: MapObserver<T>,
+    OT: ObserversTuple<I, S>,
+    S: HasCorpus<C, I> + HasMetadata,
+    Z: Evaluator<E, EM, I, S>,
+{
+    map_observer_name: String,
+    #[allow(clippy::type_complexity)]
+    phantom: PhantomData<(C, E, EM, I, O, OT, S, T, Z)>,
+}
+
+// The number of times we run the program in the calibration stage
+const CAL_STAGE_MAX: usize = 8;
```

Comment:
Could there be a benefit for having this value dynamically as struct member, instead of constant?